src/inventory_ui.h

```diff
@@ -85,28 +82,85 @@ class inventory_entry
         }
 };
 
-class inventory_column
+class inventory_selector_preset
 {
     public:
-        inventory_column() :
-            entries(),
-            mode( navigation_mode::ITEM ),
-            active( false ),
-            multiselect( false ),
-            custom_colors( false ),
-            selected_index( 1 ),
-            page_offset( 0 ),
-            entries_per_page( 1 ) {}
+        inventory_selector_preset();
+
+        virtual bool is_shown( const item &it ) const;
+        virtual bool is_shown( const item_location &location ) const;
+
+        virtual bool is_enabled( const item &it ) const;
+        virtual bool is_enabled( const item_location &location ) const;
```

Comment:
As @Coolthulhu pointed out, the majority of consumers would want `item`. I'd spare them the conversion and leave it as it is.